examples/nextjs-with-typescript/src/Link.tsx

```diff
@@ -17,7 +17,17 @@ interface NextLinkComposedProps
 
 export const NextLinkComposed = React.forwardRef<HTMLAnchorElement, NextLinkComposedProps>(
   function NextLinkComposed(props, ref) {
-    const { to, linkAs, replace, scroll, shallow, prefetch, locale, ...other } = props;
+    const {
+      to,
+      linkAs,
+      replace,
+      scroll,
+      shallow,
+      prefetch,
+      legacyBehavior = true,
```

Comment:
Is it s temporary fix? 

What will we do when `.legacyBehavior` be removed in next versions of NextJs ?

Reply:
As I mention in my comment on #34898 a proper fix would be doing something like `<MuiLink component={NextLink}/>` but that isn't directly compatible with nextjs 12

`legacyBehaviour` is recommended by next themselves to be used with function or custom components and won't most likely be removed for quite a while as it would be a breaking change
https://nextjs.org/docs/api-reference/next/link